src/healthkit.cs

```diff
@@ -171,6 +202,54 @@ public interface HKCategorySample {
 		HKCategorySample FromType (HKCategoryType type, nint value, NSDate startDate, NSDate endDate, [NullAllowed] HKDevice device, [NullAllowed] NSDictionary<NSString,NSObject> metadata);
 	}
 
+	[Watch (3,0), iOS (10,0)]
+	[BaseType (typeof(HKSample))]
+	[DisableDefaultCtor] // NSInvalidArgumentException Reason: The -init method is not available on HKDocumentSample
+	interface HKDocumentSample
+	{
+		[Export ("documentType", ArgumentSemantic.Strong)]
+		HKDocumentType DocumentType { get; }
+	}
+
+	[NoWatch, iOS (10,0)]
+	[BaseType (typeof(HKDocumentSample))]
+	[DisableDefaultCtor] // NSInvalidArgumentException Reason: The -init method is not available on HKCDADocumentSample
+	interface HKCDADocumentSample
```

Comment:
You have to provide the native name if it does not match (e.g. `NSUrl`) like:

    [BaseType (typeof(HKDocumentSample), Name = "HKCDADocumentSample")]
    interface HKCdaDocumentSample {